master/buildbot/status/web/forms.py

```diff
@@ -34,9 +35,56 @@ def content(self, request, cxt):
     def getChild(self, path, req):
         if path == "forceBuild":
             return ForceBuildDialogPage()
+        elif path == "rebuild":
+            return RebuildDialogPage()
 
+class BuildDialogPage(HtmlResource):
+    def _decodeArgs(self, request, encoding):
+        args = request.args.copy()
+        for name, argl in args.iteritems():
+            if '_branch' in name:
+                args[name] = [ self.decodeFromURL(arg, encoding) for arg in argl ]
+        return args
 
-class ForceBuildDialogPage(HtmlResource):
+    def _getSlaves(self, builder_status):
+        slaves = [s for s in builder_status.getSlaves() if s.isConnected()]
+        slaves = sorted(slaves, key=attrgetter('friendly_name'))
+        return slaves
+
+    def _getBranches(self, args, request):
+        encoding = getRequestCharset(request)
+        branches = [b.decode(encoding) for b in args.get("branch", []) if b]
```

Comment:
same as before just return it

Reply:
no idea - most of this is just methods refactored/extracted from the existing code to avoid duplication. But looking at it does remove the default case if "branch" does not exists. Then b is [] which is False and then does not get decoded? :)

Reply:
Nah if `branch` doesn't exists in 'for in` statement then there is no any elements. It's more like if there is in branches a branch which has None or '' value. But ok I get it.